als-server/shared/src/main/scala/org/mulesoft/als/server/workspace/WorkspaceManager.scala

```diff
@@ -148,11 +157,14 @@ class WorkspaceList(environmentProvider: EnvironmentProvider,
                     projectConfigurationProvider: ProjectConfigurationProvider,
                     editorConfiguration: EditorConfiguration,
                     telemetryProvider: TelemetryProvider,
-                    val allSubscribers: List[BaseUnitListener],
+                    val allSubscribers: List[AstListener[_] with WorkspaceContentListener],
                     logger: Logger,
                     configurationProvider: ConfigurationProvider) {
 
-  def subscribers: List[BaseUnitListener] = allSubscribers.filter(_.isActive)
+  def subscribers: List[AstListener[_] with WorkspaceContentListener] = allSubscribers.filter(_.isActive)
```

Comment:
see comment above, his filter should be a better implementation than the strict match by type